### ISSUE TITLE
semi and no-extra-semi report duplicate errors

### ISSUE THREAD

**Issue by btmills** [eslint/eslint#2207]
```js
// test.js
while (true) ;
;;
function foo() { };
```

```js
// config.json
{
    "rules": {
        "no-extra-semi": 2,
        "semi": [2, "never"]
    }
}
```

```sh
$ eslint --reset --no-eslintrc -c config.json test.js

test.js
  2:13  error  Unnecessary semicolon  no-extra-semi
  2:14  error  Extra semicolon        semi
  3:0   error  Unnecessary semicolon  no-extra-semi
  3:1   error  Extra semicolon        semi
  3:1   error  Unnecessary semicolon  no-extra-semi
  3:2   error  Extra semicolon        semi
  4:18  error  Unnecessary semicolon  no-extra-semi
  4:19  error  Extra semicolon        semi

✖ 8 problems (8 errors, 0 warnings)
```

**Comment by nzakas** [comment]
Seems like we should talk about this one. What's the right fix? Either we don't need `no-extra-semi` or we remove the check from `semi`?

**Comment by btmills** [comment]
I'm in favor of removing the check from semi. To me the purpose of the two rules is different: semi in "never" mode prohibits unnecessary semicolons, whereas no-extra-semi prohibits empty statements. Semicolons are always necessary in empty statements, so it seems that semi is overstepping its bounds by disallowing empty statements. Further, the only way right now to forbid empty statements with semi in "always" mode is to keep no-extra-semi.

**Comment by nzakas** [comment]
Sounds good.
